jerboa/finetune.py

```diff
@@ -166,9 +162,13 @@ def train(
 
     prompter = Prompter(prompt_template_name)
 
-    device_map = "auto"
+    # Device_map is not compatible with CPU training
+    device = 'cuda' if torch.cuda.is_available() else 'cpu'
+    device_map = "auto" if device == "cuda" else None
     world_size = int(os.environ.get("WORLD_SIZE", 1))
-    ddp = world_size != 1
+
+    # Only apply if we have more than one GPU, not if we have no GPU
+    ddp = world_size > 1
```

Comment:
not important, i would rather implement a utilitity function called `is_distribuetd` to determine if it is DDP, an easy way is : 

```python
  if 'WORLD_SIZE' in os.environ:
      return int(os.environ['WORLD_SIZE']) > 1
```

Reply:
This returns a key error if we have no gpu